### ISSUE TITLE
[Broken Website] Google Cloud Platform stopped working

### ISSUE THREAD

**Issue by steebchen** [darkreader/darkreader#7503]
# Broken Website Report

GCP stopped working recently, it used to work great but now just doesn't work at all anymore.

## Website Address
https://console.cloud.google.com/logs/query
https://console.cloud.google.com/traces/overview

## Steps To Reproduce
Most (all?) of Google Cloud Console seems to be affected, just visit any product like Cloud SQL, Logs, Traces, Billing, etc. 

## Expected Behavior
Dark mode.

## Actual Behavior
The page is white and there is no dark mode. Switching between dynamic/filter mode doesn't change anything.

## System Information

- OS: MacOS
- Browser: Safari
- Dark Reader Version: 1.4.4 from apple store

**Comment by mattwelke** [comment]
Google Cloud console has always been hit and miss for me with the extension. Sometimes it works, and sometimes it's like the toggle has no effect, even if I refresh the page. Today, it's working. That's why I came here to look for whether an issue or PR related to it had been closed.

**Comment by steebchen** [comment]
For me it's still not working :/

**Comment by Gusted** [comment]
@alexanderby Given it's a safari issue.

**Comment by mattwelke** [comment]
I think https://github.com/darkreader/darkreader/issues/660 may have something to do with this. There appears to be a rule that the apex domain rules overrides the rules for any subdomain. So if the apex domain is added to Dark Reader's global dark list (meant to track sites that it shouldn't allow itself to be enabled for), then the subdomains are considered dark too, and the extension can't be enabled for subdomains. Maybe `google.com` has been added and removed periodically from this global dark list?

**Comment by steebchen** [comment]
FWIW it works right now on the latest version. Not sure if it's just a coincidence but it didn't work for a while but now it does again for me.

**Comment by mattwelke** [comment]
I think it has something to do with how subdomains work with the extension. I can consistently reproduce the issue by enabling the extension on `cloud.google.com` (because I want to read the docs with the dark theme). Then, when I try to enable the extension on `console.cloud.google.com`, it doesn't work. If I disable the extension on `cloud.google.com`, then I can enable the extension on `console.cloud.google.com`.

**Comment by Gusted** [comment]
> I think it has something to do with how subdomains work with the extension. I can consistently reproduce the issue by enabling the extension on `cloud.google.com` (because I want to read the docs with the dark theme). Then, when I try to enable the extension on `console.cloud.google.com`, it doesn't work. If I disable the extension on `cloud.google.com`, then I can enable the extension on `console.cloud.google.com`.

That's another issue, that's being tracked by #147

**Comment by mattwelke** [comment]
@Gusted Thanks for the link. I think we should close this issue because I've demonstrated that Dark Reader does work for console.cloud.google.com, as long as you aren't using exclude lists in a way that prevents you from enabling Dark Reader on it.

**Comment by unitysipu** [comment]
> > If I disable the extension on `cloud.google.com`, then I can enable the extension on `console.cloud.google.com`.
You just saved my life, thank you! My 600 nit screen burns my eyes on the cloud console.

**Comment by mattwelke** [comment]
True, it's a good workaround for the lack of more granular subdomain controls that we have right now.

I ended up not using this extension much on console.cloud.google.com because it seemed to slow it down too much. I wish more GUIs for developers would have dark theme support built in. :( I find that this extension tends to only be performant on static HTML pages. Works great for docs.

**Comment by bershanskiy** [comment]
> I find that this extension tends to only be performant on static HTML pages. Works great for docs.

Dark Reader should work well on dynamic pages as well, if you could report a site which slows down with Dark Reader enabled, please open a dedicated issue. Sometimes, multiple extensions running on the same page break each other. Do you have any other extensions installed?

**Comment by mattwelke** [comment]
Only two extensions I use are Dark Reader and uBlock Origin. On web apps, I disable uBlock Origin because they don't typically display ads and adblocking sometimes causes functionality problems.

I can open an issue for Google Cloud console. Not sure how I'd quantify the problem though. So far, it just "feels slow". I'm not familiar with ways to measure the performance of websites. I could maybe record a video where it captures mouse clicks, to demonstrate things like delay between clicking and the web app doing what it's supposed to do.
